### PR TITLE
[SYCL][NFC] Refactor CG type versioning

### PR DESCRIPTION
**Pull request by romanovvlad**
Removed dedicated `KERNEL_V1` value, start using setType and getType
which hide manipulations with the version.

### CODE REVIEW

File: sycl/source/handler.cpp [hunk 1]
```diff
@@ -127,9 +127,8 @@ event handler::finalize() {
   }
 
   unique_ptr_class<detail::CG> CommandGroup;
-  switch (MCGType) {
+  switch (static_cast<detail::CG::CGTYPE>(getType())) {
```
Comment (alexbatashev) [inline sycl/source/handler.cpp:130]: Why do you need `static_cast` here? `getType` returns CGTYPE, doesn't it?
```suggestion
  switch (getType()) {
```

File: sycl/source/handler.cpp [hunk 2]
```diff
@@ -204,6 +203,10 @@ event handler::finalize() {
                         PI_INVALID_OPERATION);
   }
 
+  if (!CommandGroup)
+    throw runtime_error("Internal Error. Command group cannot be constructed.",
```
Comment (romanovvlad) [inline sycl/source/handler.cpp:207]:
```suggestion
    throw sycl::runtime_error("Internal Error. Command group cannot be constructed.",
```

File: sycl/include/CL/sycl/handler.hpp [hunk 3]
```diff
@@ -2212,7 +2222,9 @@ class __SYCL_EXPORT handler {
   string_class MKernelName;
   /// Storage for a sycl::kernel object.
   shared_ptr_class<detail::kernel_impl> MKernel;
-  /// Type of the command group, e.g. kernel, fill.
+  /// Type of the command group, e.g. kernel, fill. Can also encode version.
+  /// Use getType and setType methods to access this variable unless
+  /// manupulations with version are required
```
Comment (sergey-semenov) [inline sycl/include/CL/sycl/handler.hpp:2227]:
```suggestion
  /// manipulations with version are required
```

**Review by alexbatashev — APPROVED** [review]
LGTM

**Review by sergey-semenov — APPROVED** [review]

**Pull request merged**